File: Web/Angular/src/app/interfaces/checkList.ts
import {User} from './user';

class CheckList {
  id: number;
  name: string;
  taskId: number;
  createdById: number;
  createdTime: string;
  createdBy: User;
  changedById: number;
  changedBy: User;
  changedTime: string;
  items: CheckListItem[];
}

class CheckListItem {
  id: number;
  name: string;
  isChecked: boolean;
}

export {CheckList, CheckListItem}

File: Entity/Models/CheckList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entity.Models
{
public class CheckList
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public int TaskID { get; set; }
        public Nullable<int> ChangedBy { get; set; }
        public Nullable<int> CreatedBy { get; set; }
        public Nullable<System.DateTime> ChangedTime { get; set; }
        public Nullable<System.DateTime> CreatedTime { get; set; }

        public virtual Task Task { get; set; }
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<CheckListItem> CheckListItems { get; set; }
    }
}
